data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexType.java

```diff
@@ -1,7 +1,38 @@
 package com.amazon.dataprepper.plugins.sink.opensearch.index;
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
 public enum IndexType {
-    TRACE_ANALYTICS_RAW,
-    TRACE_ANALYTICS_SERVICE_MAP,
-    CUSTOM;
+    TRACE_ANALYTICS_RAW("trace-analytics-raw"),
+    TRACE_ANALYTICS_SERVICE_MAP("trace-analytics-service-map"),
+    CUSTOM("custom");
+
+    private final String value;
+
+    // Reverse-lookup map for getting a IndexType from a name
+    private static final Map<String, IndexType> STRING_TO_INDEX_TYPE_MAP = new HashMap<>();
+
+    static {
+        Arrays.stream(IndexType.values())
+                .forEach(indexType -> STRING_TO_INDEX_TYPE_MAP.put(indexType.value, indexType));
+    }
+
+    IndexType(final String value){
+        this.value = value;
+    }
+
+    String getValue(){
+        return value;
+    }
+
+    static Optional<IndexType> getByValue(final String value) {
```

Comment:
Please add a unit test for this class so that we can verify this behavior in isolation.

Reply:
By the way, you can write unit tests that verify all different enum scenarios by using JUnit 5's parameterized testing with `@EnumSource`. https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-EnumSource

Reply:
I added IndexTypeTests.java in new revision with test cases. 
However, OpenSearch Sink Plugin depends on JUnit4 now, so it does not have `@EnumSource`. 
Upgrading all test cases to Junit5 takes time, not in this PR scope.

Reply:
I look forward to making that update in the future!